scripts/git-hooks/check_cli_doc.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+
+top_level=`git rev-parse --show-toplevel`
+str_functions_rst="src/doc/cli_manual/functions.rst"
+str_method_doc_C="src/visitpy/common/MethodDoc.C"
+str_method_doc_h="src/visitpy/common/MethodDoc.h"
+method_doc_staged=false
+
+# Look to see if functions.rst is staged. If it is, then run function_to_method_doc and look for
+# changes in MethodDoc.h/C. Enforce coupling between staged changes to functions.rst and resulting
+# changes in MethodDoc.h/C
+for staged in $(git diff --cached --name-only)
+do
+  if [ $staged == $str_functions_rst ]; then
+    method_doc_staged=true
+    $top_level/src/doc/functions_to_method_doc.py $top_level
+      
+    mdc_staged=false
+    mdh_staged=false
+    for unstaged in $(git diff --name-only)
+    do
+      # functions.rst should not have staged and unstaged changes because the
+      # MethodDoc.h/C will not correlate with the commiting functions.rst
+      if [ $unstaged == $str_functions_rst ]; then
+        echo "ERROR: Cannot have both staged and unstaged changes to functions.rst."
+        echo "Please stage all changes to functions.rst or stash unstaged changes."
+        exit 1
+      elif [ $unstaged == $str_method_doc_C ]; then
+        mdc_staged=true
+      elif [ $unstaged == $str_method_doc_h ]; then
+        mdh_staged=true
+      fi
+    done
+    
+    # Output error messages and exit
+    if [ $mdc_staged == true ] && [ $mdh_staged == true ]; then
+      echo "ERROR: Your changes to functions.rst will modify MethodDoc.h and MethodDoc.C, but you are not staging those."
+      echo "Please also stage MethodDoc.h and MethodDoc.C and try again."
+      exit 1
+    elif [ $mdc_staged == true ]; then
+      echo "ERROR: Your changes to functions.rst will modify MethodDoc.C, but you are not staging it."
+      echo "Please also stage MethodDoc.C and try again."
+      exit 1
+          
+    elif [ $mdh_staged == true ]; then
+      echo "ERROR: Your changes to functions.rst will modify MethodDoc.h, but you are not staging it."
+      echo "Please also stage MethodDoc.h and try again."
```

Comment:
Shouldn't changes to `MethodDoc.[Ch]` be done and staged automagically by a hook?

Reply:
I'm hesitant to auto-stage changes in commits that the developer may not know about. For example, I tend to check the build before I make any commits (unless I know I'm temporarily breaking build), so if the pre-commit hook automatically adds changes that I did not expect that accidentally break build, I'd would be really confused if I then submitted a PR and the CI tests broke. I think it's best to require the developer to explicitly decide what goes into a commit.

I could add some instructions about adding in the error that might help them get a better idea of where MethodDoc.[Ch] is located. I could also call a `git status` from the script so they can see the changes.

Reply:
I suppose we could add a make target to re-gen `MethodDoc.[Ch]` when `functions.rst` is newer. At least it would  then appear in the *modified but not yet staged list* during `git status`.

I am fine with not auto-committing it. I like the idea of producing an error message with a little more instruction. We can avoid that for now and see if it becomes necessary and add it if so.

Reply:
> I suppose we could add a make target to re-gen `MethodDoc.[Ch]` when `functions.rst` is newer. At least it would then appear in the _modified but not yet staged list_ during `git status`.

I'm sorry that I wasn't clear above; this is actually exactly what happens. If the pre-commit hook detects staged changes in functions.rst, then it automatically runs functions_to_method_doc.py ([line 16](https://github.com/visit-dav/visit/pull/3540/files#diff-3357be862abcee4bd240e23de72906d5R16)), thus updating MethodDoc.[Ch]. If the developer already did that, then there will be no changes to MethodDoc.[Ch] and the commit will go through. However, if the developer did not run the update script, then the commit will be rejected, and a `git status` will show that MethodDoc.[Ch] has unstaged changes (from the auto-running in the hook), and they could easily add those change and check the build before committing again.